File: src/main/java/no/fintlabs/access/control/config/RestClientConfig.java
package no.fintlabs.access.control.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Configuration
public class RestClientConfig {

    @Value("${fint.metamodel-url}")
    private String metamodelUrl;

    @Bean
    public RestClient restClient() {
        return RestClient.builder()
                .baseUrl(metamodelUrl)
                .build();
    }

}

File: src/main/java/no/fintlabs/access/control/config/RestClientConfig.kt
package no.fintlabs.access.control.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
open class RestClientConfig(
    @Value("\${fint.metamodel-url}")
    private val metamodelUrl: String
) {
    @Bean
    open fun restClient(): RestClient {
        return RestClient.builder()
            .baseUrl(metamodelUrl)
            .build()
    }
}
